### DISCUSSION TITLE
[Enhancement]: General UI ideas

### DISCUSSION THREAD

**Discussion by VasyaSovari** [ciderapp/Cider#658] [Feature Request]
### The feature you would like

Hi there. I've just returned to Apple Music after switching to Spotify for a couple of years cos it worked so much better on Linux. This is _easily_ the best app I've found for AM on Arch. Some ideas for improvement:

1. Show 'Heart' indicator inline for each song, and make it a clickable toggle. Makes it vastly easier to manage playlists. Right-clicking to check each song individually is a massive pita and I edit playlists constantly

2. Enable album tools (Love, add to playlist etc) from album view, rather than just Artist view

3. Check whether songs are already on a playlist before adding them (I'm aware this may not be possible on AM at all) 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by KillerParticles** [comment]
The first part of number 1 has recently been added to Cider, and soon will be togglable. 

Number 3 should already be a thing and should be working properly.

**Reply by VasyaSovari** [reply]
Excellent work, much appreciated
1. Noted, and it looks great. I have had a few cases where the Heart doesn't show on songs I've previously 'Loved', but it's not a huge deal to correct that manually as I go
3. Yes, this has helped enormously. The number of albums I've added to a playlist 2 or 3 times is frankly embarrassing
